src/components/Checkbox.js

```diff
@@ -82,6 +82,11 @@ class Checkbox extends React.Component {
             return;
         }
 
+        // Make sure checkbox get unfocused when it's unselected
+        if (this.state.isFocused && this.props.isChecked) {
+            this.onBlur();
+        }
```

Comment:
This code is never firing on the web for me.

Reply:
This will fire in the edge case I mentioned here.

> @parasharrajat there was another edge case with onPressOut, when you press the checkbox at very edge of the border, it won't get called, so the outline still green, that's why I still need handle it inside onClick. Here's the video about it:
> 
>  Screen.Recording.2022-12-01.at.00.06.41.mov

Reply:
Yeah, testing it on Safari. I didn't see that edge case on mweb chrome.

Reply:
Ok, it seems that it firing on the Safari simulator. we can clean this up a bit. 

So you want to unfocus when the checkbox is deselected. If so, this code is not guaranteed to work correctly. Because the parent component changes the checked state which is not in control of the Checkbox component, we can't say that the onPress call changed the checked state. 

But this code will not have any side effects as we are also setting the focused border when `this.props.checked` is true. 

I think we can clean this up like so with a comment. 


// Make sure the checkbox gets unfocused when it's about to be unselected

```js
const wasChecked = this.props.isChecked;
if (this.state.isFocused && wasChecked) {
 this.onBlur();
 }
```

Reply:
@parasharrajat  Cool. I updated the PR.